apollo-router/src/axum_http_server_factory.rs

```diff
@@ -2508,6 +2635,121 @@ Content-Type: application/json\r
         }
     }
 
+    #[tokio::test]
+    async fn it_makes_sure_same_listenaddrs_are_accepted() {
+        let configuration = Configuration::fake_builder().build();
+
+        init_with_config(MockSupergraphService::new(), configuration, MultiMap::new()).await;
+    }
+
+    #[tokio::test]
+    #[should_panic(
+        expected = "Failed to create server factory: DifferentListenAddrsOnSamePort(127.0.0.1, 0.0.0.0, 4000)"
+    )]
+    async fn it_makes_sure_different_listenaddrs_but_same_port_are_not_accepted() {
+        let configuration = Configuration::fake_builder()
+            .graphql(
+                Graphql::fake_builder()
+                    .listen(SocketAddr::from_str("127.0.0.1:4000").unwrap())
+                    .build(),
+            )
+            .sandbox(
+                Sandbox::fake_builder()
+                    .listen(SocketAddr::from_str("0.0.0.0:4000").unwrap())
+                    .build(),
+            )
+            .build();
+
+        init_with_config(MockSupergraphService::new(), configuration, MultiMap::new()).await;
+    }
+
+    // TODO: axum just panics here.
+    // While this is ok for now, we probably want to check it ourselves and return a meaningful error
+    #[tokio::test]
+    #[should_panic(
+        expected = "Invalid route: insertion failed due to conflict with previously registered route: /"
+    )]
+    async fn it_makes_sure_extra_endpoints_cant_use_the_same_listenaddr_and_path() {
+        let configuration = Configuration::fake_builder()
+            .graphql(
+                Graphql::fake_builder()
+                    .listen(SocketAddr::from_str("127.0.0.1:4000").unwrap())
+                    .build(),
+            )
+            .build();
+        let endpoint = service_fn(|_req: transport::Request| async move {
+            Ok::<_, BoxError>(
+                http::Response::builder()
+                    .body("this is a test".to_string().into())
+                    .unwrap(),
+            )
+        })
+        .boxed();
+
+        let mut mm = MultiMap::new();
+        mm.insert(
+            SocketAddr::from_str("127.0.0.1:4000").unwrap().into(),
```

Comment:
Just a NIT, could we use different port than `4000` which is the default one and when tested locally it conflicts with my dev instance of the router :D

Reply:
would 4010 work for everyone?